pallets/committee-management/src/impls.rs

```diff
@@ -345,18 +347,49 @@ impl<T: Config> Pallet<T> {
         committee
     }
 
+    pub(crate) fn calculate_underperforming_finalizers(session_id: SessionIndex) {
+        let CurrentAndNextSessionValidators {
+            current: SessionValidators { finalizers, .. },
+            ..
+        } = CurrentAndNextSessionValidatorsStorage::<T>::get();
+
+        let underperformed_session_count_threshold =
+            FinalityBanConfig::<T>::get().underperformed_session_count_threshold;
+        let minimal_expected_performance =
+            FinalityBanConfig::<T>::get().minimal_expected_performance;
+
+        let is_underperforming = |score| score > minimal_expected_performance;
+
+        let finalizers_perf = T::AbftScoresProvider::scores_for_session(session_id)
+            .map(|score| score.points)
+            .unwrap_or(vec![minimal_expected_performance; finalizers.len()])
+            .into_iter()
+            .map(is_underperforming);
+
+        // TODO make sure ordering is correct on validators list
```

Comment:
This TODO probably shouldn't be here.

If it's about the ABFT order agreeing with the one from the pallets then it should, it should just be the source of that order, unless there is something unnecessarily shuffling it around somewhere in between.

Reply:
I've removed the comment.